mlir/lib/Conversion/AIRToAIESchedulingUtils.cpp

```diff
@@ -1009,6 +1015,12 @@ template <typename T> int air::foundInVector(T item, std::vector<T> vec) {
   return index;
 }
 
+template <typename T>
+void air::push_back_if_unique(SmallVector<T> &vec, T entry) {
+  if (std::find(vec.begin(), vec.end(), entry) == vec.end())
+    vec.push_back(entry);
+}
+
```

Comment:
aka `llvm::SmallSet` ?

Reply:
Make sense. Will replace them all. Thanks.

Reply:
Hmm looks like Value cannot be used as key to SmallSet, because `mlir::Value` isn't invocable by `std::less`:
```
error: static assertion failed due to requirement 'std::__is_invocable<std::less<mlir::Value> &, const mlir::Value &, const mlir::Value &>{}': comparison object must be invocable with two arguments of key type
```

Reply:
> Hmm looks like Value cannot be used as key to SmallSet, because `mlir::Value` isn't invocable by `std::less`:
> 
> ```
> error: static assertion failed due to requirement 'std::__is_invocable<std::less<mlir::Value> &, const mlir::Value &, const mlir::Value &>{}': comparison object must be invocable with two arguments of key type
> ```

doh. Sorry for the distraction.

Reply:
Not at all, as the `push_back_if_unique` has also been used elsewhere on `Operation *`, which works with `SmallSet`. I have a PR to come very soon.